src/test/java/htsjdk/samtools/cram/encoding/CramRecordWriterReaderTest.java

```diff
@@ -0,0 +1,81 @@
+package htsjdk.samtools.cram.encoding;
+
+import htsjdk.samtools.ValidationStringency;
+import htsjdk.samtools.cram.encoding.reader.CramRecordReader;
+import htsjdk.samtools.cram.io.BitInputStream;
+import htsjdk.samtools.cram.io.DefaultBitInputStream;
+import htsjdk.samtools.cram.io.ExposedByteArrayOutputStream;
+import htsjdk.samtools.cram.structure.CompressionHeader;
+import htsjdk.samtools.cram.structure.CramCompressionRecord;
+import htsjdk.samtools.cram.structure.Slice;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class CramRecordWriterReaderTest extends CramRecordTestHelper {
+    private CramCompressionRecord read(final byte[] dataBytes,
+                                       final CompressionHeader header,
+                                       final int refId,
+                                       final Map<Integer, InputStream> inputMap) throws IOException {
+        try (final ByteArrayInputStream is = new ByteArrayInputStream(dataBytes)) {
+            final BitInputStream bis = new DefaultBitInputStream(is);
+
+            final CramRecordReader reader = new CramRecordReader(bis, inputMap, header, refId, ValidationStringency.DEFAULT_STRINGENCY);
+            final CramCompressionRecord recordToRead = new CramCompressionRecord();
+            reader.read(recordToRead);
+            return recordToRead;
+        }
+    }
+
+    private List<CramCompressionRecord> initRTRecords() {
+        final List<CramCompressionRecord> initialRecords = new ArrayList<>();
+
+        // note for future refactoring
+        // createRecord() calls Sam2CramRecordFactory.createCramRecord()
+        // which is the only way to set a record's readFeatures (except via read codec)
+        // which would otherwise be null
+
+        CramCompressionRecord record = createRecord();
```

Comment:
It's kind of funny that this just has 1 record.   Better than no records though!

Reply:
added a few more